### PR TITLE
Adds a Fox Plushie

### PR DESCRIPTION
**Pull request by KingOfThePing**
This PR adds different coloured fox plushies to the loadout. You can either select the standard red one or a random coloured one:

If you poke it, it laughs.

### Asset Licenses
The following assets that **have not** been created by myself are included in this PR:

| Path | Original Author | License |
| --- | --- | --- |
| icons/obj/toy.dmi (foxplushie) | Unknown (TauCetiStation) | GNU3 |

### CODE REVIEW

**Comment by KingOfThePing** [comment]
!review

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1075,6 +1075,34 @@
 	item_state = "[colorvar]squid"
 	desc = "A small, cute and loveable squid friend. This one is in [colorvar]."
 
+//Fox Plushies
+
+
+/obj/item/toy/plushie/fox
```
Comment (SleepyGemmy) [inline code/game/objects/items/toys.dm:1081]:
```suggestion
// Fox Plushies
/obj/item/toy/plushie/fox
```

Comment (Greenjoe12345) [reply code/game/objects/items/toys.dm:1081]: Couldn't we just have all the colors selectable in the loadout
Rather then having red or leaving it to chance if you get the color you want

**Comment by Greenjoe12345** [comment]
Couldn't we just have all the colors selectable in the loadout
Rather then having red or leaving it to chance if you get the color you want

**Comment by niennab** [comment]
Not sure this really matches the server's style.

**Comment by KingOfThePing** [comment]
> Couldn't we just have all the colors selectable in the loadout Rather then having red or leaving it to chance if you get the color you want

I guess this is reasonable

**Comment by KingOfThePing** [comment]
You can now select all colours in the loadout or a random one.

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1133,6 +1133,96 @@
...
+	phrase = "Kekekeke!"
+	var/active = 0.0
+	var/colorvar = "red"
+
+/obj/item/toy/plushie/fox/random/Initialize()
+	. = ..()
+	colorvar = pick("red","black","marble","blue","orange","coffee","pink","purple","crimson")
+	icon_state = "[colorvar]fox"
+	item_state = "[colorvar]fox"
+	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is in [colorvar]."
```
Comment (SleepyGemmy) [inline code/game/objects/items/toys.dm:1223]:
```suggestion
//Fox Plushies

/obj/item/toy/plushie/fox
	name = "fox plushie"
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is red."
	icon = 'icons/obj/toy.dmi'
	icon_state = "redfox"
	item_state = "redfox"
	phrase = "Kekekeke!"

/obj/item/toy/plushie/fox/black
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is black."
	icon_state = "blackfox"
	item_state = "blackfox"

/obj/item/toy/plushie/fox/marble
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is white."
	icon_state = "marblefox"
	item_state = "marblefox"

/obj/item/toy/plushie/fox/blue
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is blue."
	icon_state = "bluefox"
	item_state = "bluefox"

/obj/item/toy/plushie/fox/orange
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is the classic orange."
	icon_state = "orangefox"
	item_state = "orangefox"

/obj/item/toy/plushie/fox/coffee
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is coffee-cream coloured."
	icon_state = "coffeefox"
	item_state = "coffeefox"

/obj/item/toy/plushie/fox/pink
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is pink."
	icon_state = "pinkfox"
	item_state = "pinkfox"

/obj/item/toy/plushie/fox/purple
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is purple."
	icon_state = "purplefox"
	item_state = "purplefox"

/obj/item/toy/plushie/fox/crimson
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is crimson."
	icon_state = "crimsonfox"
	item_state = "crimsonfox"

/obj/item/toy/plushie/fox/random
	icon_state = "redfox"
	item_state = "redfox"
	var/active = 0.0
	var/colorvar = "red"

/obj/item/toy/plushie/fox/random/Initialize()
	. = ..()
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is in [colorvar]."
	icon_state = "[colorvar]fox"
	item_state = "[colorvar]fox"
	colorvar = pick("red", "black", "marble", "blue", "orange", "coffee", "pink", "purple", "crimson")
```

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1133,6 +1133,96 @@
...
+	item_state = "crimsonfox"
+	phrase = "Kekekeke!"
+/obj/item/toy/plushie/fox/random
+	name = "fox plushie"
+	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is red."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "redfox"
+	item_state = "redfox"
+	phrase = "Kekekeke!"
+	var/active = 0.0
```
Comment (SleepyGemmy) [inline code/game/objects/items/toys.dm:1215]: unused/leftover var?

File: code/modules/client/preference_setup/loadout/loadout_general.dm [hunk 4]
```diff
@@ -290,6 +290,16 @@
...
+	plushies["plushie, red fox"] = /obj/item/toy/plushie/fox
+	plushies["plushie, black fox"] = /obj/item/toy/plushie/blackfox
+	plushies["plushie, marble fox"] = /obj/item/toy/plushie/marblefox
+	plushies["plushie, blue fox"] = /obj/item/toy/plushie/bluefox
+	plushies["plushie, orange fox"] = /obj/item/toy/plushie/orangefox
+	plushies["plushie, coffee fox"] = /obj/item/toy/plushie/coffeefox
+	plushies["plushie, pink fox"] = /obj/item/toy/plushie/pinkfox
+	plushies["plushie, purple fox"] = /obj/item/toy/plushie/purplefox
+	plushies["plushie, crimson fox"] = /obj/item/toy/plushie/crimsonfox
+	plushies["plushie, random fox"] = /obj/item/toy/plushie/fox/random
```
Comment (SleepyGemmy) [inline code/modules/client/preference_setup/loadout/loadout_general.dm:302]:
```suggestion
	plushies["plushie, red fox"] = /obj/item/toy/plushie/fox
	plushies["plushie, black fox"] = /obj/item/toy/plushie/fox/black
	plushies["plushie, marble fox"] = /obj/item/toy/plushie/fox/marble
	plushies["plushie, blue fox"] = /obj/item/toy/plushie/fox/blue
	plushies["plushie, orange fox"] = /obj/item/toy/plushie/fox/orange
	plushies["plushie, coffee fox"] = /obj/item/toy/plushie/fox/coffee
	plushies["plushie, pink fox"] = /obj/item/toy/plushie/fox/pink
	plushies["plushie, purple fox"] = /obj/item/toy/plushie/fox/purple
	plushies["plushie, crimson fox"] = /obj/item/toy/plushie/fox/crimson
	plushies["plushie, random fox"] = /obj/item/toy/plushie/fox/random
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm (hunk 3 above)
Comment (KingOfThePing) [reply code/game/objects/items/toys.dm:1215]: leftover myes

File: code/game/objects/items/toys.dm [hunk 5]
```diff
@@ -1133,6 +1133,59 @@
...
+	item_state = "pinkfox"
+/obj/item/toy/plushie/fox/purple
+	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is purple."
+	icon_state = "purplefox"
+	item_state = "purplefox"
+/obj/item/toy/plushie/fox/crimson
+	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is crimson."
+	icon_state = "crimsonfox"
+	item_state = "crimsonfox"
+/obj/item/toy/plushie/fox/random
```
Comment (SleepyGemmy) [inline code/game/objects/items/toys.dm:1176]:
```suggestion
/obj/item/toy/plushie/fox
	name = "fox plushie"
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is red."
	icon = 'icons/obj/toy.dmi'
	icon_state = "redfox"
	item_state = "redfox"
	phrase = "Kekekeke!"

/obj/item/toy/plushie/fox/black
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is black."
	icon_state = "blackfox"
	item_state = "blackfox"

/obj/item/toy/plushie/fox/marble
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is white."
	icon_state = "marblefox"
	item_state = "marblefox"

/obj/item/toy/plushie/fox/blue
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is blue."
	icon_state = "bluefox"
	item_state = "bluefox"

/obj/item/toy/plushie/fox/orange
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is the classic orange."
	icon_state = "orangefox"
	item_state = "orangefox"

/obj/item/toy/plushie/fox/coffee
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is coffee-cream coloured."
	icon_state = "coffeefox"
	item_state = "coffeefox"

/obj/item/toy/plushie/fox/pink
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is pink."
	icon_state = "pinkfox"
	item_state = "pinkfox"

/obj/item/toy/plushie/fox/purple
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is purple."
	icon_state = "purplefox"
	item_state = "purplefox"

/obj/item/toy/plushie/fox/crimson
	desc = "A small, very soft and handsome fox, ready to receive your cuddles. This one is crimson."
	icon_state = "crimsonfox"
	item_state = "crimsonfox"

/obj/item/toy/plushie/fox/random
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**